### ISSUE TITLE
"source" load def startup file with variable name

### ISSUE THREAD

**Issue by schmitzli** [nushell/nushell#2990]
**Describe the bug**
cannot source a file when the filename is a variable

**To Reproduce**
```
▶ let STARTUP_FILE = C:\Users\Stefan\Workspace\nu\startup.nu
▶ source $STARTUP_FILE
error: Type Error
  ┌─ shell:4:8
  │
4 │ source $STARTUP_FILE
  │        ^^^^^^^^^^^^^ Expected a filepath constant, found $STARTUP_FILE
```
**Expected behavior**
using the variable should work as it is with open or echo 

**Configuration (please complete the following information):**

Run `version | pivot` and paste the output to show OS, features, etc.

```
> version | pivot
 0 │ version            │ 0.26.0
 1 │ build_os           │ windows-x86_64
 2 │ rust_version       │ rustc 1.48.0 (7eac88abb 2020-11-16)
 3 │ rust_channel       │ stable-x86_64-pc-windows-msvc (default)
 4 │ cargo_version      │ cargo 1.48.0 (65cbdd2dc 2020-10-14)
 5 │ pkg_version        │ 0.26.0
 6 │ build_time         │ 2021-01-29 21:13:14
 7 │ build_rust_channel │ release
 8 │ features           │ clipboard-cli, ctrlc, default, directories, dirs, ichwh, ptree, rustyline, term, trash,
   │                    │ uuid, which, zip
```

**Add any other context about the problem here.**

**Comment by LhKipp** [comment]
The relevant code is: 

https://github.com/nushell/nushell/blob/b052d524da3ed5cbc85d955c60a4694a0fb8f55d/crates/nu-parser/src/parse.rs#L1779-L1787

I was wondering whether we should introduce a second command like: import (which would do what `source` is currently doing) and let source behave as one might think. @jonathandturner

**Comment by jonathandturner** [comment]
@LhKipp yeah, we could go that route. Right now `source` works like a static import, so it may make sense to split those up.

**Comment by Mrfiregem** [comment]
This would be helpful for sourcing a directory with many `.nu` files

**Comment by hustcer** [comment]
We need this feature badly, Thanks.

**Comment by fellnerse** [comment]
I would like this as well!

**Comment by JCallicoat** [comment]
I am new to nu so I might be doing something silly!

I was trying to play with splitting up the config file like mentioned in #4662 just to see if I liked it. I get the following in 0.59.0:

```
〉let config_dir = (build-string $nu.home-path "/.config/nushell")
〉let git_completions = ($config_dir | path join "completions/git.nu")
〉source $git_completions
Error: nu::parser::file_not_found (link)

  × File not found
   ╭─[entry #20:1:1]
 1 │ source $git_completions
   ·        ────────┬───────
   ·                ╰── File not found: $git_completions
```

I think this is the same problem as this issue, but with the new engine I believe it comes from treating `name_expr` as a literal filename here https://github.com/nushell/nushell/blob/a6a96b29cbadf42e2fb0908e71947689e8861669/crates/nu-parser/src/parse_keywords.rs#L1558-L1564

**Comment by jmoore34** [comment]
If you're looking to reload your config file, here is a keyboard shortcut you can add in 0.59:

config.nu
```
let config = {
...
  keybindings: [
    {
        name: reload_config
        modifier: alt
        keycode: char_r
        mode: emacs
        event: [
            { edit: { cmd: clear } }
            { edit: { cmd: insertString value: $"source ($nu.config-path)" } }
            { send: Enter }
        ]
    }
  ]
```

**Comment by 89z** [comment]
Given example doesnt work with current version:

~~~
PS D:\Desktop\nushell-0.60.0> .\nu.exe
Error: nu::shell::unsupported_config_value (link)

  × Unsupported config value
     ╭─[C:\Users\Steven\AppData\Roaming\nushell\config.nu:223:1]
 223 │         event: [
 224 │             { edit: { cmd: insertString value: $"source ($nu.config-path)" } }
     ·             ─────────────────────────────────┬────────────────────────────────
     ·                                              ╰── expected reedline EditCommand, got {cmd: insertstringvalue: source c:\users\steven\appdata\roaming\nushell\config.nu}
~~~

**Comment by jmoore34** [comment]
The event syntax was tweaked a little bit since then. Here is an updated example from the [0.60 release post](https://www.nushell.sh/blog/2022-03-22-nushell_0_60.html):

```
{
    name: reload_config
    modifier: none
    keycode: f5
    mode: emacs
    event: {
      send: executehostcommand,
      cmd: $"source '($nu.config-path)'"
    }
}
```

**Comment by 89z** [comment]
I just want to insert the literal string `cd ..`, the new syntax doesnt seem to allow that...

**Comment by 89z** [comment]
This seems to do it:

~~~
{
   mode: emacs
   modifier: control
   keycode: up
   event: [
      {
         edit: insertstring
         value: 'cd ..'
      } {
         send: enter
      }
   ]
}
~~~

**Comment by alechp** [comment]
Noticed the previous example for 0.44 where `char_r` was being used, but thought that might've changed in 0.60

I was a little confused with the Char codes because of what keybindings and [crossterm docs](https://docs.rs/crossterm/0.14.0/crossterm/event/enum.KeyCode.html) say
> ![Screen Shot 2022-03-25 at 6 41 54 PM](https://user-images.githubusercontent.com/3058939/160219757-74ed34f0-8d98-4d11-ac7e-14215e48c4d6.png)

When I tried referencing keybindings, seemed like change was necessary. But obviously that failed
![Screen Shot 2022-03-25 at 6 40 54 PM](https://user-images.githubusercontent.com/3058939/160219795-04c518ff-65cc-4720-bcc3-69dee891b67a.png)
![Screen Shot 2022-03-25 at 6 40 48 PM](https://user-images.githubusercontent.com/3058939/160219796-605c1152-639b-43b9-869c-a733e4bcee48.png)


What's strange is that the modifier key *does* look exactly like what's presented in keybindings. However, using the old modifier format failed:
![Screen Shot 2022-03-25 at 6 47 12 PM](https://user-images.githubusercontent.com/3058939/160219992-9baf849a-5263-4d0b-bce3-e8bbdba545cc.png)

This is the modified keybinding I'm using now:
```
{
        name: reload_config
        modifier: Control
        # keycode: Char('a')
        keycode: char_a
        mode: emacs
        event: {
          send: executehostcommand,
          cmd: $"source '($nu.config-path)'; echo 'nushell config updated'"
        }
    }
```

Adding that to your keybindings, works as expected:
![Screen Shot 2022-03-25 at 6 44 58 PM](https://user-images.githubusercontent.com/3058939/160219892-0590f47d-c596-4ad3-9d2f-a57d27ee730f.png)

> I added the "NEWENV" *before* running the script just to confirm changes the script update works as expected

**Comment by JCallicoat** [comment]
@jmoore34 My use case was a little different, I wasn't trying to reload the config, I was trying to split different parts of the config out to different files and then source them from the main config file to play with your idea in #4662 

I wanted to try splitting out git completions to a different file just to play around with it and tried this from the main config (where the git completions were in `$HOME/.config/nushell/completions/git.nu`):

```
〉let config_dir = (build-string $nu.home-path "/.config/nushell")
〉let git_completions = ($config_dir | path join "completions/git.nu")
〉source $git_completions
```

But I ran into this current issue were it seems that the expression passed to the source command is treated as a literal string for the filename and is not evaluated as a regular nushell expression first.
